pam/return.go

```diff
@@ -1,54 +1,41 @@
 package main
 
+import (
+	"github.com/msteinert/pam"
+)
+
 // Various signalling return messaging to PAM.
 
-// pamSuccess signals PAM module to return PAM_SUCCESS and Quit tea.Model.
+type pamReturnStatus interface {
+	Status() pam.StatusError
+	Error() string
+}
+
+// TODO: Remove this when we've tests coverage that ensures this is the case.
+// ensure that [pam.NewTransactionError] implements [pamReturnStatus].
+var _ pamReturnStatus = pam.NewTransactionError(pam.ErrAbort, nil)
+
+// pamStatus signals PAM module to return with provided pam.Success and Quit tea.Model.
 type pamSuccess struct {
+	pam.TransactionError
 	brokerID string
 }
 
-// String returns the string of pamSuccess.
-func (err pamSuccess) String() string {
-	return ""
+// newPamSuccess returns a new pamSuccess.
+func newPamSuccess(brokerID string) pamReturnStatus {
+	return pamSuccess{nil, brokerID}
 }
 
-// pamIgnore signals PAM module to return PAM_IGNORE and Quit tea.Model.
+// pamIgnore signals PAM module to return pam.Ignore and Quit tea.Model.
 type pamIgnore struct {
+	pam.TransactionError
```

Comment:
same, here, what’s the value of returning the error rather than a message at the stage? I don’t see the error value being returned as an error but then transformed as a string (but I may be missing something here).